### ISSUE TITLE
[SOLVED] Dummy Cam in Stamps

### ISSUE THREAD

**Issue by melMass** [adrianpueyo/Stamps#6]
**SOLVED**: 

TLDR, use `world_matrix` instead of `matrix`

---

Hi Adrian,

Thanks a lot for Stamps, this with W_Hotbox completely changed my work for the better.

I'm adding a few scripts to W_Hotbox and I want to use a Stamped camera as one input to this WHB script, I found that the class is simply 'Camera', but the matrix is not transferred properly.

My setup:
![image](https://user-images.githubusercontent.com/7041726/121027133-027acc00-c7a7-11eb-8c33-bf9425687d82.png)

```python
camera1["matrix"].getValue()
# Result: 
# [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0]

cam10["matrix"].getValue()
# Result: 
# [0.9999542236328125, 0.005690377205610275, 0.007688807789236307, -0.006094004493206739, -0.005690541118383408, 0.9999837875366211, -5.52225174033083e-07, 1.2598254670592723e-06, -0.00768868625164032, -4.320127482060343e-05, 0.9999704360961914, -0.0015999485040083528, 0.0, 0.0, 0.0, 1.0]
```
I could traverse the tree upward to find the camera, but I'm pretty sure the idea of the DummyCam is not having to deal with that.

Thanks

**Comment by melMass** [comment]
Ohhh wait, by default it's not using dummy cam I think!

Here @ 3:20:
https://vimeo.com/showcase/6323584/video/362949593

I'm testing it further and will report

**Comment by melMass** [comment]
Yes, DummyCam on its own is picking up the right matrix:

![image](https://user-images.githubusercontent.com/7041726/121029571-fc85ea80-c7a8-11eb-85a9-7a0779967798.png)

But I checked my Stamps config (I'm using the default one from 1.1) and DummyCam is used for Camera, but the info still states "Camera"? 

![image](https://user-images.githubusercontent.com/7041726/121029961-54bcec80-c7a9-11eb-9ab4-c91761dddd80.png)

**Comment by adrianpueyo** [comment]
Good to hear DummyCam's working for you!
So DummyCam's class is actually Camera. It's just a Camera that has expressions on the projection knobs.
http://www.nukepedia.com/gizmos/3d/dummycam
Please confirm everything's working as expected!
Thanks,
Adrian

**Comment by melMass** [comment]
Thanks, DummyCam is working on its own, but not when used with Stamps apparently, at least the matrix:

```
display wireframe
selectable true
file_menu Axis/File
snap_menu Axis/Snap
xform_order SRT
rot_order ZXY
translate {0 0 0}
rotate {0 0 0}
scaling {1 1 1}
uniform_scale 1
skew {0 0 0}
pivot {0 0 0}
useMatrix false
matrix {
    {1 0 0 0}
    {0 1 0 0}
    {0 0 1 0}
    {0 0 0 1}
  }
projection_mode perspective
focal {65.34520085}
haperture {36}
vaperture {24}
near {0.1}
far {10000}
win_translate {0 0}
win_scale {1 1}
winroll {0}
focal_point {2}
fstop {16}
the_cam node153e53800.
the_frame 1.03334e+06
d_projection_mode perspective
d_focal 50
d_haperture 24.576
d_vaperture 18.672
d_near 0.1
d_far 10000
d_win_translate {0 0}
d_win_scale {1 1}
d_winroll 0
d_focal_point 2
d_fstop 16
lockCallbacks 0
toReconnect false
title CAM_10
postageStamp_show false
advanced_reconnection 0
anchor Anchor_275ee3001
auto_reconnect_by_title false
```

**Comment by adrianpueyo** [comment]
Weird! Could you make a simplified .nk script that still produces the issue and attach it? Also which version of nuke and os are you using?

**Comment by melMass** [comment]
Yes sorry, here it is 
[stamps_camera_issue.nk.zip](https://github.com/adrianpueyo/Stamps/files/6609785/stamps_camera_issue.nk.zip)

**Config**:
- Nuke 13.0v2
- macOS 10.15 (Catalina)

**Comment by melMass** [comment]
![image](https://user-images.githubusercontent.com/7041726/121035244-f9d9c400-c7ad-11eb-8237-e55424923fba.png)

**Comment by adrianpueyo** [comment]
Hey, I think it's working for me. Please be aware the knob you're looking for isn't matrix but global_matrix. It's a bit confusing :) 
The knob matrix refers only to the local matrix, which indeed needs to be the identity, not to modify the parent stamp in any way.

**Comment by melMass** [comment]
Yes!! Thanks, it's `world_matrix`!
What is strange is that `matrix` on `Camera3` returns the `world_matrix`
